File: Programmers/Lv.1/햄버거 만들기/133502.java
import java.util.*;

class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;
        
        List<Integer> recipe = Arrays.asList(1, 2, 3, 1);
        List<Integer> list = new ArrayList<>();
        for (int i : ingredient) list.add(i);
        
        for (int j = 0; j < list.size() - 3; j++) {
            if (list.size() < 4) break;
            
            if (recipe.equals(list.subList(j, j + 4))) {
                list.subList(j, j + 4).clear();
                answer++;
                j = Math.max(-1, j - 4);
            }
        }
        
        return answer;
    }
}

File: Programmers/Lv.1/햄버거 만들기/133502.js
function solution(ingredient) {
    let answer = 0;
    
    for (let i = 0; i < ingredient.length; i++) {
        const arr = ingredient.slice(i, i + 4);
        
        if (arr[0] === 1 && arr[1] === 2 && arr[2] === 3 && arr[3] === 1) {
            answer++;
            ingredient.splice(i, 4);
            i = Math.max(-1, i - 3);
        }
    }
    
    return answer;
}
